packages/opentelemetry-resources/src/platform/browser/detectors/BrowserDetector.ts

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { diag } from '@opentelemetry/api';
+import { SemanticResourceAttributes } from '@opentelemetry/semantic-conventions';
+import { Detector, Resource, ResourceDetectionConfig } from '../../..';
+import { ResourceAttributes } from '../../../types';
+
+/**
+ * BrowserDetector will be used to detect the resources related to browser.
+ */
+class BrowserDetector implements Detector {
+  async detect(config?: ResourceDetectionConfig): Promise<Resource> {
+    const browserResource: ResourceAttributes = {
+      [SemanticResourceAttributes.PROCESS_RUNTIME_NAME]: 'browser',
+      [SemanticResourceAttributes.PROCESS_RUNTIME_DESCRIPTION]: 'Web Browser',
+      [SemanticResourceAttributes.PROCESS_RUNTIME_VERSION]: window.navigator.userAgent
```

Comment:
Does it really make sense to use the entire UA here? In the not too distant future the UA is going to get frozen (so it won't identify the current browser version or runtime). It would be better to crack the UA and/or use the `navigator.userAgentData` [getHighEntropyValues](https://developer.mozilla.org/en-US/docs/Web/API/NavigatorUAData/getHighEntropyValues) async function -- but being an async it has it's own issues about when this data is populated

Reply:
Looks like its inside  the spec: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/resource/semantic_conventions/process.md#javascript-runtimes

Reply:
Please feel free to drop a comment at open-telemetry/opentelemetry-specification#2290. But even if the experimental `navigator.userAgentData` is available with the latest browsers, we have to fall back to userAgent if it is not available right? I was thinking that maybe we can suggest parsing the UA header to get the versions, but that parsing result is merely suggestive, IIUC.